src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,200 @@
+
+The statistical complexity is exclusively used in combination with the related information measure
+(entropy).
+`complexity(c::StatisticalComplexity, x)` returns only the statistical complexity.
+The entropy can be accessed as a `Ref` value of the struct as
+```julia
+x = randn(100)
+c = StatisticalComplexity
+compl = complexity(c, x)
+entr = c.entr_val[]
+```
+To obtain both the entropy and the statistical complexity together as a `Tuple`, use the wrapper
+`entropy_complexity`(@ref).
+
+[^Rosso2007]: Rosso, O. A. et al. (2007). Distinguishing Noise from Chaos.
+    Physical Review Letters 99, no. 15: 154102. https://doi.org/10.1103/PhysRevLett.99.154102.
+[^Rosso2013]: Rosso, O. A. (2013) Generalized Statistical Complexity: A New Tool for Dynamical Systems.
+
+"""
+Base.@kwdef struct StatisticalComplexity{E, D, H} <: ComplexityEstimator
+    dist::D = JSDivergence()
+    est::E = SymbolicPermutation()
+    entr::H = Renyi()
+    entr_val::Base.RefValue{Float64} = Ref(0.0)
+end
+
+function complexity(c::StatisticalComplexity, x)
+    (; dist, est, entr) = c
+
+    p = probabilities(est, x)
+
+    H_q = entropy_normalized(entr, est, x)
+
+    L = total_outcomes(est, x)
+
+    # calculate distance between calculated distribution and uniform one
+    D_q = evaluate(dist, vec(p), fill(1.0/L, size(p)))
+
+    # generate distribution with just one filled bin
+    deterministic = zeros(size(p))
+    deterministic[1] = 1
+
+    D_max = evaluate(dist, deterministic, fill(1.0/L, size(p)))
+    C_q = D_q / D_max * H_q
+    c.entr_val[] = H_q
+
+    return C_q
+end
+
+"""
+    entropy_complexity(c::StatisticalComplexity, x)
+
+Calculate both the `StatisticalComplexity`(@ref) and corresponding entropy
+and return both as a `Tuple`.
+"""
+function entropy_complexity(c::StatisticalComplexity, x)
+    compl = complexity(c, x)
+   return (c.entr_val[], compl)
+end
+
+function complexity(c::StatisticalComplexity, p::Probabilities)
+    (; dist, est, entr) = c
+
+    L = total_outcomes(est)
+    norm = log(entr.base, L)
+    H_q = entropy(entr, p) / norm
+
+    # calculate distance between calculated distribution and uniform one
+    D_q = evaluate(dist, vec(p), fill(1.0/L, size(p)))
+
+    # generate distribution with just one filled bin
+    deterministic = zeros(size(p))
+    deterministic[1] = 1
+
+    D_max = evaluate(dist, deterministic, fill(1.0/L, size(p)))
+    C_q = D_q / D_max * H_q
+    c.entr_val[] = H_q
+
+    return C_q
+end
+
+function fill_probs_k!(p, prob_params, L, i, k)
+    probs = vec(p)
+    probs[1] = prob_params[k] # why set first element here if overwriting it in the loop below?
+    # if we know that p has sufficient length relative to L and i,
+    # @inbounds can save some computation time by skipping bounds checking.
+    @inbounds for j = 1:(L - i)
+        probs[j] = (1 - prob_params[k]) / (L - i)
+    end
+end
+
+linearpermissiverange(start; stop, length) = length==1 ? (start:start) : range(start, stop=stop, length=length)
```

Comment:
I'm not sure why tis exists. It shouldn't. length=1 works with `range` as well.

Reply:
What does `range(0, stop=1/6, length=1)` mean though? What would you expect this to return? 

Maybe it is easier if you tell use what you want we tell you what code would do what you want. The approach you have now is type unstable and hence bad practice.

Reply:
okay; just do `linearpermissiverange = collect( ..... )` for now. With `....` what you have already.